### ISSUE TITLE
CIDER doesn't work if the file is outside of project root

### ISSUE THREAD

**Issue by Inc0n** [clojure-emacs/cider#3268]
## Introduction 
It works ~in `inf-clojure` but~ not in CIDER. So I think to some other people, this may be considered as a feature request, but for me it is a bug. I wanted to do REPL based clojure development and wanted to switch to CIDER instead, because of how much more features rich CIDER is. I set up a `~/.clojure/*` project with stuff like `add-libs` `quick-bench` from other libraries imported. This would give me a reasonable lisp REPL experience, similar to that of commonlisp. Now the actual development will happen in some other directory. And this is how this "bug" or feature request stemmed from.

EDIT: it doesn't work in inf-clojure also, but there is inf-clojure-eval-buffer.
## Actual behavior

No responses from `C-x C-e` `C-M-x` or `C-c C-l` from the file outside project. Which are CIDER functions of eval last sexp, eval defun, and load file respectively.

By no responses, means the sexp or file is not loaded or evaluated into the CIDER session, and echo area and repl shows no signs of sexp being processed.

## Expected behavior

The sexp from the file, or the whole file will be successfully evaluted into CIDER session or REPL.

## Steps to reproduce the problem

Start a project from `~/project-a` and jack in in that project then open a clojure file from `~/project-b`.
Interactions from files under project a works fine. Nothing will work in project b.
 
## Environment & Version information
Clojure CLI version 1.11.1.1189

### CIDER version information

*Include here the version string displayed when
CIDER's REPL is launched. Here's an example:*

```
;; CIDER 1.6.0-snapshot (package: 20221028.1025), nREPL 1.0.0
;; Clojure 1.11.1, Java 19
```

### Lein / Clojure CLI version
Clojure 1.11.1, Java 19

### Emacs version
28.2

### Operating system

Mac Monterey M1

### JDK distribution
```
openjdk version "19" 2022-09-20
OpenJDK Runtime Environment Homebrew (build 19)
OpenJDK 64-Bit Server VM Homebrew (build 19, mixed mode, sharing)
```

**Comment by bbatsov** [comment]
By no responses do you mean the outside files are not attached to the current session? (see https://docs.cider.mx/cider/usage/managing_connections.html)

Some debug logs would be useful as well to understand better what's happening in your case.

**Comment by Inc0n** [comment]
> By no responses do you mean the outside files are not attached to the current session? (see https://docs.cider.mx/cider/usage/managing_connections.html)

Yes you are right, edited. 
And I gave it a good thorough read, and this section was helpful to resolve the problem in my case: https://docs.cider.mx/cider/usage/managing_connections.html#context-links.

**TLDR**: use `sesman-link-with-buffer` on this buffer/file that is outside the project directory.

P.S. interestingly `cider-connect-sibling-clj` didn't work.

**Comment by Inc0n** [comment]
It seems that it was inf-clojure which adds itself to `clojure-mode-hook` was the culprit all along....
